radix-engine/src/model/module.rs

```diff
@@ -22,6 +23,12 @@ impl KernelModule {
     }
 }
 
+impl KernelModule {
+    pub fn collect_events(&mut self) -> Vec<TrackedEvent> {
```

Comment:
Since this change makes events a responsibility of the modules, shouldn't this be a method in the `Module` trait?

Reply:
Not quite yet. I think the abstraction here isn't quite fully fleshed out yet so would rather postpone how this interface looks like until we better understand the right abstraction.